fix(cellml): skip variables that do not have a valid observable

bilalshaikh42 opened the pull request:
this change skips over variables in the cellML that have been labeled as `"public_interface"="in"` as these are not exposed by OpenCor after simulation. No information is lost from the reports, as these variables have a connection to another component that contains the same variable, with the public interface set to "out". Accessing the variable through that component works fine. This addresses the issues seen in importing the physiome projects.

bilalshaikh42 committed:
feat(cellml): add flag to skip over non observable variables in cellml files

add a flag to  skips over variables in the cellML that have been labeled as `"public_interface"="in"` or `"private_interface"="in"` as these are not exposed by OpenCor after simulation. No information is lost from the reports, as these variables have a connection to another component that contains the same variable, with the public interface set to "out". Accessing the variable through that component works fine.

bilalshaikh42 commented:
@jonrkarr I think some of the issues we are seeing with the invalid observables are addressed with this change and might be a distinct issue from https://github.com/opencor/opencor/issues/2612

jonrkarr commented:
This seems fine. I don't think this is addresses https://github.com/opencor/opencor/issues/2612.

jonrkarr reviewed (changes requested):
Please fix the failing tests

bilalshaikh42 commented:
@jonrkarr I changed this to keep the default behavior but added a flag to skip the non-observable values. That way the original tests/behavior is preserved, and I added a test case for when the flag is set to true. 
This should be good to merge, but let me know if I should just have the method skip over the non-observable variables by default

bilalshaikh42 commented:
Also, I am not an admin on the biosimulators organization for sonar cloud, so I cannot configure the scanning to ignore the tests files for duplication errors.  If you add me I can make the relevant changes, or we can just ignore it.

jonrkarr commented:
> Also, I am not an admin on the biosimulators organization for sonar cloud, so I cannot configure the scanning to ignore the tests files for duplication errors. If you add me I can make the relevant changes, or we can just ignore it.

You're already an owner of the organization. I don't see permissions for SonarCloud specifically. Where can I configure that?

bilalshaikh42 commented:
https://sonarcloud.io/organizations/biosimulators/members should have an option to "configure synchronization" which will let you sync the permissions to the github organization
